Yoast/whip
# Tests: use PSR-4

### Tests: ensure filenames match the class name [1]

... and that the test class name reflects the name of the class under test.

### Tests: ensure filenames match the class name [2]

### Tests: use PSR-4

* Move the existing unit tests to a `Unit` subdirectory and ensure subdirectory names use "ProperCase".
* Ensure all test class file names match the name of the test class.
* Namespace all test files and remove file docblocks.
    There is one exception: the `WPCoreFunctionsMock` file: this contains mocks of WP native functions which need to be in the global namespace.
* Remove prefixes from the test class names which are not needed in namespaced classes (`[Whip_]DismissStorageMock`) and where relevant, update references to those classes in the test files.
* Use fully qualified global functions and constants in the namespaced test files.
* Change the Composer `autoload-dev` directive to use PSR-4 autoloading.
* Update the `phpunit.xml.dist` file to match.
* Update the `.phpcs.xml.dist` file to match.
